tests/testcases.docopt

```diff
@@ -950,8 +950,35 @@ local options: --baz
 other options:
  --egg
  --spam
--not-an-option-
 
-"""
+r"""
 $ prog --baz --egg
 {"--foo": false, "--baz": true, "--bar": false, "--egg": true, "--spam": false}
+
+#
+# docopt 0.6.2 compatibility: Blank line in options section
+# https://github.com/jazzband/docopt-ng/issues/33
+#
+r"""Usage: prog [options]
+
+    -h, --help
+    -v, --verbose        be verbose
+
+    -i, --interactive    interactive picking
+    -p, --patch          select hunks interactively
+"""
+@pytest.mark.xfail
```

Comment:
This should be `$ prog --interactive`?

Reply:
This will always pass because no test case is actually generated.

Really, we need to fix conftest.py
https://github.com/jazzband/docopt-ng/blob/88496b8f765f3197048151b8a9a7d95cf35e34f0/tests/conftest.py#L14-L27

Something like
```
if len(cases) == 0:
    raise DocoptTestException(f"Each example must have at least one test case starting with '$'. Example: {doc}")
````

Reply:
Ah, oops. That `@pytest.mark.xfail` line is a mistake too — I was playing with allowing marking cases with pytest marks (to allow xfailing cases, or tagging cases to run a subset) but then I pulled that out, but forgot to remove that line. And seems like I butchered the lines around there somehow.

I can put in a check for that, yep.

Reply:
I added some error handling to conftest based on your suggestion, should be more robust against these kind of mistakes now! It also reports details of which test fails if there's a JSON syntax error in an test too.